lib/cli/do_volapi.js

```diff
@@ -0,0 +1,393 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright (c) 2016, Joyent, Inc.
+ */
+
+/*
+ * The 'sdcadm experimental volapi' CLI subcommand.
+ */
+
+var assert = require('assert-plus');
+var util = require('util'),
+    format = util.format;
+var vasync = require('vasync');
+
+var common = require('../common');
+var errors = require('../errors');
+var DownloadImages = require('../procedures/download-images').DownloadImages;
+var shared = require('../procedures/shared');
+
+var NB_MBS_IN_GB = 1024;
+var NFS_SHARED_VOLUMES_PACKAGES_NAME_PREFIX = 'nfs_shared_volume_';
+// Sizes are in GBs
+var NFS_SHARED_VOLUMES_PKG_SIZES = [10, 20, 30, 40, 50, 60, 70, 80, 90, 100,
+    200, 300, 400, 500, 600, 700, 800, 900, 1000];
+
+/**
+ * This is the template used for creating package objects for NFS shared
+ * volumes. The size and owners' uuid are setup at runtime.
+ */
+var NFS_SHARED_VOLUMES_PACKAGE_TEMPLATE = {
+    active: true,
+    cpu_cap: 100,
+    max_lwps: 1000,
+    max_physical_memory: 1024,
+    max_swap: 1024,
+    vcpus: 1,
+    version: '1.0.0',
+    zfs_io_priority: 20,
+    default: false
+};
+
+/**
+ * Adds a package using PAPI client "papiClient" for shared NFS volumes of size
+ * "size" GBs. Calls "callback" when done with an error object and the newly
+ * created package as parameters.
+ */
+function addSharedVolumePackage(cli, packageSettings, callback) {
+    assert.object(cli, 'cli');
+    assert.object(packageSettings, 'packageSettings');
+    assert.number(packageSettings.size, 'size');
+    assert.arrayOfString(packageSettings.owner_uuids,
+        'packageSettings.owner_uuids');
+    assert.func(callback, 'callback');
+
+    var papiClient = cli.sdcadm.papi;
+
+    var packageName = NFS_SHARED_VOLUMES_PACKAGES_NAME_PREFIX
+        + packageSettings.size;
+    cli.progress(util.format('Creating "%s" package', packageName));
+
+    var newPackage = {
+        name: packageName,
+        quota: packageSettings.size * NB_MBS_IN_GB,
+        owner_uuids: packageSettings.owner_uuids
+    };
+
+    common.objCopy(NFS_SHARED_VOLUMES_PACKAGE_TEMPLATE, newPackage);
+
+    papiClient.add(newPackage, callback);
```

Comment:
make this function be a no-op of the package already exists. Perhaps I missed that.